diff -u -N -r gcc-3.4.4-orig/config.sub gcc-3.4.4/config.sub
--- gcc-3.4.4-orig/config.sub	2004-02-22 22:44:23.000000000 +0800
+++ gcc-3.4.4/config.sub	2005-11-04 23:24:53.000000000 +0800
@@ -119,7 +119,7 @@
 maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
   nto-qnx* | linux-gnu* | linux-dietlibc | linux-uclibc* | uclinux-uclibc* | uclinux-gnu* | \
-  kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* | storm-chaos* | os2-emx* | rtmk-nova*)
+  kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* | storm-chaos* | os2-emx* | rtmk-nova* | delirium* )
     os=-$maybe_os
     basic_machine=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`
     ;;
@@ -145,7 +145,7 @@
 	-convergent* | -ncr* | -news | -32* | -3600* | -3100* | -hitachi* |\
 	-c[123]* | -convex* | -sun | -crds | -omron* | -dg | -ultra | -tti* | \
 	-harris | -dolphin | -highlevel | -gould | -cbm | -ns | -masscomp | \
-	-apple | -axis)
+	-apple | -axis )
 		os=
 		basic_machine=$1
 		;;
@@ -1163,7 +1163,7 @@
 	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* \
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
-	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly*)
+	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* | -delirium*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
@@ -1529,6 +1529,9 @@
 			-vos*)
 				vendor=stratus
 				;;
+			-delirium*)
+				vendor=shikita
+				;;
 		esac
 		basic_machine=`echo $basic_machine | sed "s/unknown/$vendor/"`
 		;;
diff -u -N -r gcc-3.4.4-orig/configure.in gcc-3.4.4/configure.in
--- gcc-3.4.4-orig/configure.in	2005-03-09 01:31:40.000000000 +0800
+++ gcc-3.4.4/configure.in	2005-11-05 10:58:16.000000000 +0800
@@ -351,6 +351,12 @@
     skipdirs="$skipdirs zlib fastjar target-libjava target-boehm-gc target-zlib"
     noconfigdirs="$noconfigdirs target-libgloss"
     ;;
+  *-*-delirium)
+    skipdirs="target-libiberty ${libgcj} target-libstdc++-v3 target-libf2c target-librx"
+    skipdirs="$skipdirs target-libobjc target-examples target-groff target-gperf"
+    skipdirs="$skipdirs zlib fastjar target-libjava target-boehm-gc target-zlib"
+    noconfigdirs="$noconfigdirs $(libgcj) target-libffi target-qthreads"
+    ;;
   *-*-netbsd*)
     # Skip some stuff on all NetBSD configurations.
     noconfigdirs="$noconfigdirs target-newlib target-libiberty target-libgloss"
@@ -515,6 +521,7 @@
     ;;
   i[[3456789]]86-*-coff | i[[3456789]]86-*-elf)
     noconfigdirs="$noconfigdirs ${libgcj}"
+    #noconfigdirs="$noconfigdirs target-libffi target-qthreads"
     ;;
   i[[3456789]]86-*-freebsd* | i[[3456789]]86-*-kfreebsd*-gnu)
     noconfigdirs="$noconfigdirs target-newlib target-libgloss"
diff -u -N -r gcc-3.4.4-orig/gcc/config/i386/delirium.h gcc-3.4.4/gcc/config/i386/delirium.h
--- gcc-3.4.4-orig/gcc/config/i386/delirium.h	1970-01-01 08:00:00.000000000 +0800
+++ gcc-3.4.4/gcc/config/i386/delirium.h	2005-11-05 12:57:25.000000000 +0800
@@ -0,0 +1,4 @@
+#undef TARGET_VERSION
+#define TARGET_VERSION fprintf (stderr, " (i386 DeLirIuM ELF)");
+#undef TARGET_C99_FUNCTIONS  
+#define TARGET_C99_FUNCTIONS 0
diff -u -N -r gcc-3.4.4-orig/gcc/config/i386/t-delirium gcc-3.4.4/gcc/config/i386/t-delirium
--- gcc-3.4.4-orig/gcc/config/i386/t-delirium	1970-01-01 08:00:00.000000000 +0800
+++ gcc-3.4.4/gcc/config/i386/t-delirium	2005-11-05 00:40:04.000000000 +0800
@@ -0,0 +1 @@
+CRTSTUFF_T_CFLAGS += -DARCH_i386
diff -u -N -r gcc-3.4.4-orig/gcc/config.gcc gcc-3.4.4/gcc/config.gcc
--- gcc-3.4.4-orig/gcc/config.gcc	2005-04-25 12:47:59.000000000 +0800
+++ gcc-3.4.4/gcc/config.gcc	2005-11-05 12:57:53.000000000 +0800
@@ -1219,6 +1219,11 @@
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h kaos.h i386/kaos-i386.h"
 	tmake_file="i386/t-i386elf t-svr4"
 	;;
+i[34567]86-*-delirium)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h i386/delirium.h"
+	tmake_file="i386/t-i386elf t-svr4"
+	use_fixproto=yes
+	;;
 i860-*-sysv4*)
         tm_file="${tm_file} elfos.h svr4.h i860/sysv4.h"
         tmake_file="i860/t-i860 i860/t-svr4"
